addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/internal/discovery/EnvisalinkBridgeDiscovery.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dscalarm.internal.discovery;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.NetworkInterface;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for discovering the EyezOn Envisalink 3/2DS Ethernet interface.
+ * 
+ * @author Russell Stephens - Initial Contribution
+ * 
+ */
+public class EnvisalinkBridgeDiscovery {
+	private final static Logger logger = LoggerFactory.getLogger(EnvisalinkBridgeDiscovery.class);
+
+	private DSCAlarmBridgeDiscovery dscAlarmBridgeDiscovery = null;
+	private String ipAddress;
+	private int envisalinkBridgePort = 4025;
+	private int connectionTimeout = 5;
+
+	private InetAddress localHost = null;
+	private NetworkInterface networkInterface = null;
+	private short networkPrefix = 0;
+	private long netMask;
+	private long baseIP;
+	private long firstIP;
+	private long lastIP;
+	private long numberOfIPs;
+
+	/**
+	 * Constructor.
+	 */
+	public EnvisalinkBridgeDiscovery(DSCAlarmBridgeDiscovery dscAlarmBridgeDiscovery) {
+		this.dscAlarmBridgeDiscovery = dscAlarmBridgeDiscovery;
+	}
+
+	/**
+	 * Method for Bridge Discovery.
+	 */
+	public synchronized void discoverBridge() {
+		logger.debug("Starting Envisalink Bridge Discovery.");
+
+		Socket socket = null;
+		
+		setupIP();
+		
+		logger.debug("   Local IP Address: {} - {}",localHost.getHostAddress(), ipToLong(localHost.getHostAddress()));
+		logger.debug("   Subnet:           {} - {}",longToIP(baseIP), baseIP);
+		logger.debug("   Network Prefix:   {}",networkPrefix);
+		logger.debug("   Network Mask:     {}",longToIP(netMask));
+		logger.debug("   First IP:         {}",longToIP(firstIP));
+		logger.debug("   Last IP:          {}",longToIP(lastIP));
+
+		for(long ip=firstIP; ip <= lastIP; ip++) {
+			try {
+				ipAddress = longToIP(ip);
+				if(!ipAddress.equals("10.1.32.6")) {
+				socket = new Socket();
+				socket.setReuseAddress(true);
+				socket.setReceiveBufferSize(32);
+				socket.connect(new InetSocketAddress(ipAddress, envisalinkBridgePort), connectionTimeout);
+				
+				if (socket.isConnected()) {
```

Comment:
Hm, is there again a way to make sure that it is the correct device?
The port 4025 is actually reserved for partimage by IANA: http://www.speedguide.net/port.php?port=4025, so there is a certain probability that you discover the wrong server.